### PR TITLE
feat: add TaskScheduler

### PR DESCRIPTION
**Pull request by yshrsmz**
Added TaskScheduler.

Implementation is basically the same as android-client-sdk, but there is a difference: This SDK only has foreground schedulers. Behavior, when a browser tab is not active, depends on the browser's spec.

The basic implementation should be finished now.
Remaining tasks are

- E2E tests
- example
- Release automation

### CODE REVIEW

File: test/internal/evaluation/EvaluationInteractor.spec.ts [hunk 1]
```diff
@@ -22,22 +22,21 @@ import { setupServerAndListen } from '../../utils'
...
   let server: SetupServer
+  let config: BKTConfig
   let component: DefaultComponent
   let interactor: EvaluationInteractor
   let evaluationStorage: EvaluationStorageImpl
 
   beforeEach(() => {
+    config = defineBKTConfig({
+      apiKey: 'api_key_value',
+      apiEndpoint: 'https://api.bucketeer.io',
```
Comment (cre8ivejp) [inline test/internal/evaluation/EvaluationInteractor.spec.ts:33]: Could you rename this in another PR to `apiURL` to keep consistency with the other SDKs?

**Review by cre8ivejp — APPROVED** [review]
Thank you! We are almost there!

File: test/internal/evaluation/EvaluationInteractor.spec.ts (hunk 1 above)
Comment (yshrsmz) [reply test/internal/evaluation/EvaluationInteractor.spec.ts:33]: You mean we are going to rename it everywhere?

https://github.com/bucketeer-io/android-client-sdk/blob/73508d1642da3c6db052f66a7471032aad56c269/bucketeer/src/main/kotlin/io/bucketeer/sdk/android/BKTConfig.kt#L17

Android Client SDK is currently `apiEndpoint`

Comment (yshrsmz) [reply test/internal/evaluation/EvaluationInteractor.spec.ts:33]: Ok, Swift SDK is `apiURL`

https://github.com/ca-dp/bucketeer-ios-sdk/blob/09e6a345135951fd2883bb7276d1abeac01f8efc/Bucketeer/Sources/Config.swift#L6


will do this in the next PR

**Comment by yshrsmz** [comment]
Thanks!

File: test/internal/evaluation/EvaluationInteractor.spec.ts (hunk 1 above)
Comment (cre8ivejp) [reply test/internal/evaluation/EvaluationInteractor.spec.ts:33]: Sorry, it was correct. So you don't need to rename it.

Android 2.0
https://github.com/bucketeer-io/android-client-sdk/blob/main/bucketeer/src/main/kotlin/io/bucketeer/sdk/android/BKTConfig.kt#L17

iOS 2.0
https://github.com/ca-dp/bucketeer-ios-sdk/blob/v2/Bucketeer/Sources/Public/BKTConfig.swift#L5

Comment (yshrsmz) [reply test/internal/evaluation/EvaluationInteractor.spec.ts:33]: Sure, thanks for the quick update!

**Pull request merged**